src/meta-srv/src/lease.rs

```diff
@@ -71,6 +73,32 @@ pub async fn alive_datanodes(
     .await
 }
 
+pub async fn lookup_alive_flownode_peer(
+    cluster_id: ClusterId,
+    datanode_id: u64,
+    meta_peer_client: &MetaPeerClientRef,
+    lease_secs: u64,
+) -> Result<Option<Peer>> {
+    let lease_filter = build_lease_filter(lease_secs);
+    let lease_key = FlownodeLeaseKey {
+        cluster_id,
+        node_id: datanode_id,
+    };
+    let lease_key_bytes: Vec<u8> = lease_key.clone().try_into()?;
+    let Some(kv) = meta_peer_client.get(&lease_key_bytes).await? else {
+        return Ok(None);
+    };
+    let lease_value: LeaseValue = kv.value.try_into()?;
+    if lease_filter(&lease_value) {
```

Comment:
I don't think we need to filter the results since we're finding exactly what we need. cc @WenyXu

Reply:
Same advice as `lookup_alive_datanode_peer`

Reply:
We could rename the method to `lookup_flownode_peer` and `lookup_datanode_peer` cc @WenyXu